### ISSUE TITLE
RunAt option

### ISSUE THREAD

**Issue by KrishnaPG** [logstash-plugins/logstash-input-http_poller#34]
Thank you much for sharing this plugin. It is very helpful.

One small request. Could you please consider provide some kind of `runat` option (in addition to the `interval` option). For example, we need to run report generation at, say 9pm every night. 

At present the `interval` can be set to 86400 (to run once a day) - but it is relative time (that depends on the first run or start of the logstash). For example, if we modify the config (to add another url, say) and restart the logstash, then the relative period (and hence the `interval`) gets modified and the polling times would become different. In such cases, specifying the polling time with absolute time of day would be invaluable.

Something like,
urls => {
  test1 => { runat: "21:00", url: "http://something" }
  test2 => { runat: "22:00" , url: "http://somethingelse" }
}

Or alternatively, adding support for generic cron expression would also do.

**Comment by andrewvc** [comment]
This is a great idea. I like the interface as well. I'll try and find some time to work on this.

**Comment by andrewvc** [comment]
@KrishnaPG apologies for not finding the time to work on this. We'd be glad to take a patch (with tests however!) Ideally it would integrate the cron scheduling from https://github.com/logstash-plugins/logstash-input-jdbc

**Comment by elvarb** [comment]
Have my vote on this feature, yet again would open up for more use cases

**Comment by hummingV** [comment]
@andrewvc, our team is also interested in this feature. I have pushed my implementation (with tests) in my repo. 
https://github.com/hummingV/logstash-input-http_poller/tree/rufus-scheduler

Can you take a look at that? I'll create a pull request if you want to take in the change.

**Comment by hummingV** [comment]
@andrewvc 
Good day. Our team at Adnovum is interested to have this feature and I was assigned to work on this patch. I have completed the work. It has been tested in my env and it works fine. I am just short of signing CLA (needs approval) before submitting a PR. In meantime, I am wondering whether the patch has any issue since I have not received any comment? Would you be able to comment on any concerns or improvements on this patch? I will be happy to make amendments accordingly. 

Here is the overview of the patch (commit 992a0f3):
> Use rufus scheduler similar to logstash-input-jdbc plugin. Rufus supports
> four different types of scheduling - cron, every, in and at. "every"
> option (e.g. every 5 minutes) is the same as existing Stud/interval
> scheduling. Therefore, existing 'interval' config option is removed.
> 
> rspec tests are also implemented.
> 
> Made the following interface changes to the plugin configs:
> + removed "interval" option
> + added "schedule" option. hash with key=<schedule type>,
> value=<schedule string>
>   e.g schedule => { every => "2m"}
> 

p.s: @jsvd , @ph : I am just cc-ing you since I noticed that you also contribute frequently to this plugin.

**Comment by suyograo** [comment]
@hummingV thanks for your contribution! I have one comment from your branch: Remove of `interval` will become a breaking change. We will not be able to just remove this without warning users. Can you deprecate this using `deprecate`, and then encourage users to use `schedule`? We can remove it in the next version.

**Comment by hummingV** [comment]
@suyograo, thanks for your input! I will restore the `interval` option as `deprecated`. 
But for simplicity and ease of maintenance, I suggest to restore only the API interface, not the underlying mechanism. i.e. 
`interval => 15`
is exactly equivalent to 
`schedule => { "every" => "15s" } `
i.e. it will invoke Rufus::Scheduler.every method instead of Stud.interval method. 

What do you think?

**Comment by crccheck** [comment]
Is this released? I noticed this message "You are using a deprecated config setting \"interval\" set in http_poller." This is the only reference I can find related to that. The docs for all versions https://www.elastic.co/guide/en/logstash/master/plugins-inputs-http_poller.html still say to use "interval".

**Comment by hummingV** [comment]
It looks like documentation is not updated yet:
https://github.com/logstash-plugins/logstash-input-http_poller/issues/62

You can see the usage at this file for now:
https://github.com/logstash-plugins/logstash-input-http_poller/blob/master/lib/logstash/inputs/http_poller.rb

**Comment by jordansissel** [comment]
Yeah sorry about the docs. There's a bug in our docs builder causing us to
not publish easily. We are working on it.
